ts_project does not pick up referenced project's parent tsconfig

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Oh hi there! 😄 

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 bug report

### Affected Rule

<!-- Can you pin-point one or more rules as the source of the bug? -->
<!-- ✍️edit: --> The issue is caused by the rule: `ts_project`


### Is this a regression?

<!-- Did this behavior use to work in the previous version? -->
<!-- ✍️--> Yes, the previous version in which this bug was not present was: ....


### Description

<!-- ✍️--> A clear and concise description of the problem...

let's assume project layout like below

```
- tsconfig.json
\ src
   - tsconfig.esm.json
   \ common
     - BUILD
 	 - tsconfig.json  
     - tsconfig.bzl.json 
   \ preload
     - BUILD
     - tsconfig,json
     - tsconfig.bzl.json
```

`tsconfig.json` under each package extends `\\:tsconfig.json`, `\\src:tsconfig.esm.json`. `tsconfig.bzl.json` is the actual config being used in ts_project, extends each `:tsconfig.json`.

when `\\src\preload:tsconfig.bzl` have reference to `\\src:common`, ts_project `deps` will pick up direct reference, `\\src\common:tsconfig.bzl.json` and does not pick up its extend `\\src\common:tsconfig.json`, in result compilation error occurs by missing some of config flags specified in parent config only.

## 🔬 Minimal Reproduction

- clone https://github.com/kwonoj/ts_project_composite_test, `npm test` on Linux / darwin (sandbox)
- ci output https://github.com/kwonoj/ts_project_composite_test/pull/1/checks?check_run_id=542899759

<!--
Please create and share minimal reproduction of the issue. For the purpose you can create a GitHub repository and share a link. Make sure you don't upload any confidential files.
-->

## 🔥 Exception or Error

<pre><code>
<!-- If the issue is accompanied by an exception or an error, please share it below: -->
<!-- (Please run your build with --define=VERBOSE_LOGS=1 to gather more info for us to understand the problem.) -->
<!-- ✍️-->
##[error]src/preload/tsconfig.bzl.json(4,5): error TS6306: Referenced project '/home/runner/.cache/bazel/_bazel_runner/effb5996f4ecc130424f20eff3118d8b/sandbox/linux-sandbox/4/execroot/test/src/common/tsconfig.bzl.json' must have setting "composite": true.
</code></pre>


## 🌍  Your Environment

**Operating System:**

<pre>
  <code>

  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>

  </code>
</pre>

**Rules_nodejs version:**

(Please check that you have matching versions between WORKSPACE file and `@bazel/*` npm packages.)

<pre>
  <code>

  </code>
</pre>

**Anything else relevant?**

--- a/packages/typescript/src/internal/ts_project.bzl
+++ b/packages/typescript/src/internal/ts_project.bzl
@@ -123,7 +123,7 @@ def _ts_project_impl(ctx):
                 collect_default = True,
             ),
         ),
-        _TsConfigInfo(tsconfigs = depset([ctx.file.tsconfig], transitive = [
+        _TsConfigInfo(tsconfigs = depset([ctx.file.tsconfig] + ctx.files.extends, transitive = [
             dep[_TsConfigInfo].tsconfigs
             for dep in ctx.attr.deps
             if _TsConfigInfo in dep

--- a/packages/typescript/test/ts_project/a/BUILD.bazel
+++ b/packages/typescript/test/ts_project/a/BUILD.bazel
@@ -2,7 +2,10 @@ load("@npm_bazel_typescript//:index.bzl", "ts_project")
 
 ts_project(
     composite = True,
-    extends = ["//packages/typescript/test/ts_project:tsconfig-base.json"],
+    extends = [
+        ":tsconfig-extended.json",
+        "//packages/typescript/test/ts_project:tsconfig-base.json",
+    ],
     # Intentionally not syncing this option from tsconfig, to test validator suppression
     # source_map = True,
     validate = False,

--- a/packages/typescript/test/ts_project/a/tsconfig-extended.json
+++ b/packages/typescript/test/ts_project/a/tsconfig-extended.json
@@ -1,0 +1,4 @@
+// This file is an extra hop of indirection, regression test for #1754
+{
+    "extends": "../tsconfig-base"
+}

--- a/packages/typescript/test/ts_project/a/tsconfig.json
+++ b/packages/typescript/test/ts_project/a/tsconfig.json
@@ -1,5 +1,5 @@
 {
-    "extends": "../tsconfig-base.json",
+    "extends": "./tsconfig-extended.json",
     "compilerOptions": {
         "sourceMap": true,
         // WORKAROUND https://github.com/microsoft/TypeScript/issues/37378

